sorting/SelectionSort.c

```diff
@@ -1,64 +1,44 @@
 //sorting of array list using selection sort
 #include <stdio.h>
 
-/*Displays the array, passed to this method*/
+/*To diplay array elements using display function*/
 void display(int arr[], int n){
-    
     int i;
     for(i = 0; i < n; i++){
         printf("%d ", arr[i]);
     }
-    
     printf("\n");
-    
-}
-
-/*Swap function to swap two values*/
-void swap(int *first, int *second){
-    
-    int temp = *first;
-    *first = *second;
-    *second = temp;
-    
 }
 
 /*This is where the sorting of the array takes place
  arr[] --- Array to be sorted
  size --- Array Size
  */
 void selectionSort(int arr[], int size){
-    
+    int temp;
     for(int i=0; i<size; i++) {
-        int min_index = i;
         for(int j=i+1; j<size; j++) {
-            if(arr[min_index] > arr[j]) {
-                min_index = j;
+            if(arr[i] > arr[j]) {
+                temp=arr[i];
+                arr[i]=arr[j];
+                arr[j]=temp;
```

Comment:
in selection sort, we choose the smallest element (for ascending order) between _i+1 to n_ & replace it with the i<sup>th</sup> element.
by swapping elements in every iteration, you have basically converted section sort to bubble sort.